### PR TITLE
Added test for cli.py

### PR DESCRIPTION
**Pull request by knakul853**
Fixes #

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

I added test for cli.py. I tested and it works fine :)

### CODE REVIEW

**Comment by de-sh** [comment]
@knakul853 seems like the tests succeed but aren't actually being run, maybe this is because of the absence of the __init__.py file. I am working on your submission and it seems like it doesn't even work with the addition of that file. I think we should be moving the tests folder as tests such as the pep8 one don't run on the files outside of the hydrus module(folder).

Also the Pull Request should be to branch `develop` instead of `master`.

File: tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    @classmethod
+    def tearDownClass(self):
+        """Tear down temporary database and exit utilities"""
+        self.client.__exit__(None, None, None)
+        self.doc_util.__exit__(None, None, None)
+        self.session_util.__exit__(None, None, None)
+        self.api_name_util.__exit__(None, None, None)
+        self.session.close()
+
+    def setUp(self):
```
Comment (chrizandr) [inline tests/test_cli.py]: There are no assertions/checks here. How is this a test?

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+from hydra_python_core import doc_maker
+from hydrus.samples import doc_writer_sample
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
+from hydrus.data.db_models import Base
+import click
+from click.testing import CliRunner
+from cli import startserver
+
+def test_startserver():
```
Comment (chrizandr) [inline tests/test_cli.py]: This function is never called/used. What is it here for?

File: tests/test_cli.py (hunk 1 above)
Comment (chrizandr) [inline tests/test_cli.py]: There are no assertions/checks here. How is this a test?

File: tests/test_cli.py (hunk 2 above)
Comment (de-sh) [reply tests/test_cli.py]: He meant to make it a part of the test class but it's outside the scope. Would be wonderful if you could share a good resource to learn testing as we are new to it.

Comment (knakul853) [reply tests/test_cli.py]: @chrizandr  yeah i forget to put in test class now i changed ,will make new PR

File: tests/test_cli.py (hunk 1 above)
Comment (knakul853) [reply tests/test_cli.py]: @chrizandr  what assertion i can make hear i saw another test files that does not have any aasertions for this .

File: tests/test_cli.py (hunk 2 above)
Comment (de-sh) [reply tests/test_cli.py]: @knakul853 don't make a new PR, you can always undo a commit and force push the changes.

Comment (ShravanDoda) [reply tests/test_cli.py]: @knakul853 kindly take a look at the docs
unittest -> https://docs.python.org/3/library/unittest.html
unittest.mock -> https://docs.python.org/3/library/unittest.mock.html

File: tests/test_cli.py (hunk 1 above)
Comment (de-sh) [reply tests/test_cli.py]: I will make a PR to your fork with the changes I think are worth it. Let's collaborate on it

Meanwhile can someone suggest a good resource to learn testing? :sweat_smile:

Comment (ShravanDoda) [reply tests/test_cli.py]: I think python docs for testing are pretty good

File: tests/test_cli.py (hunk 2 above)
Comment (knakul853) [reply tests/test_cli.py]: @ShravanDoda  thanks

File: tests/test_cli.py (hunk 1 above)
Comment (knakul853) [reply tests/test_cli.py]: @de-sh yep

**Comment by knakul853** [comment]
I am closing this pr and opening a new one because by mistake I opened the pull request in comparison with master branch. sorry for inconvenience.

**Pull request closed without merging**